diff --git a/Makefile.in b/Makefile.in
index 641b833..22ba4e1 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -16,7 +16,7 @@ srcdir = @srcdir@
 top_srcdir = @top_srcdir@
 top_builddir = @top_builddir@
 
-SUBDIRS = conf daemons include lib libdaemon libdm man scripts tools
+SUBDIRS = conf include libdaemon libdm scripts
 
 ifeq ("@UDEV_RULES@", "yes")
   SUBDIRS += udev
diff --git a/configure.in b/configure.in
index 70f3b78..61c9b66 100644
--- a/configure.in
+++ b/configure.in
@@ -89,7 +89,7 @@ AC_HEADER_SYS_WAIT
 AC_HEADER_TIME
 
 AC_CHECK_HEADERS([locale.h stddef.h syslog.h sys/file.h sys/time.h assert.h \
-  langinfo.h libgen.h signal.h sys/mman.h sys/resource.h sys/utsname.h \
+  libgen.h signal.h sys/mman.h sys/resource.h sys/utsname.h \
   sys/wait.h time.h], ,
   [AC_MSG_ERROR(bailing out)])
 
@@ -130,8 +130,8 @@ AC_STRUCT_TM
 
 ################################################################################
 dnl -- Check for functions
-AC_CHECK_FUNCS([ftruncate gethostname getpagesize \
-  gettimeofday memset mkdir mkfifo rmdir munmap nl_langinfo setenv setlocale \
+AC_CHECK_FUNCS([ftruncate gethostname \
+  gettimeofday memset mkdir rmdir munmap setenv setlocale \
   strcasecmp strchr strcspn strspn strdup strncasecmp strerror strrchr \
   strstr strtol strtoul uname], , [AC_MSG_ERROR(bailing out)])
 AC_FUNC_ALLOCA
@@ -1296,8 +1296,7 @@ if [[ \( "$LVM1" = shared -o "$POOL" = shared -o "$CLUSTER" = shared \
 fi
 
 ################################################################################
-AC_CHECK_LIB([pthread], [pthread_mutex_lock],
-	[PTHREAD_LIBS="-lpthread"], hard_bailout)
+AC_CHECK_FUNCS([pthread_mutex_lock], , hard_bailout)
 
 ################################################################################
 dnl -- Disable selinux
