diff --git a/Makefile.am b/Makefile.am
index 08269e2..98b54df 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -60,7 +60,7 @@ CHECK_LOCALS =
 EXTRA_DIST =
 CLEANFILES =
 
-SUBDIRS = po
+SUBDIRS =
 
 RCS_FIND_IGNORE = \( -name SCCS -o -name BitKeeper -o -name .svn -o \
 		-name CVS -o -name .pc -o -name .hg -o -name .git \) -prune -o
@@ -77,24 +77,8 @@ EXTRA_DIST += \
 
 
 include include/Makemodule.am
-include lib/Makemodule.am
 include libuuid/Makemodule.am
 include libblkid/Makemodule.am
-include libmount/Makemodule.am
-include libsmartcols/Makemodule.am
-include libfdisk/Makemodule.am
-
-include schedutils/Makemodule.am
-include text-utils/Makemodule.am
-include term-utils/Makemodule.am
-include login-utils/Makemodule.am
-include sys-utils/Makemodule.am
-include misc-utils/Makemodule.am
-include disk-utils/Makemodule.am
-
-include bash-completion/Makemodule.am
-
-include tests/Makemodule.am
 
 #
 # Don't rely on configure.ac AC_CONFIG_FILES for install paths.
diff --git a/configure.ac b/configure.ac
index bbfa54b..059756b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1811,7 +1811,6 @@ libmount/src/libmount.h
 libsmartcols/docs/Makefile
 libsmartcols/docs/version.xml
 libsmartcols/src/libsmartcols.h
-po/Makefile.in
 ])
 
 AC_OUTPUT
diff --git a/include/c.h b/include/c.h
index e423e8b..02d8c88 100644
--- a/include/c.h
+++ b/include/c.h
@@ -233,24 +233,6 @@ static inline int dirfd(DIR *d)
 #endif
 
 /*
- * MAXHOSTNAMELEN replacement
- */
-static inline size_t get_hostname_max(void)
-{
-	long len = sysconf(_SC_HOST_NAME_MAX);
-
-	if (0 < len)
-		return len;
-
-#ifdef MAXHOSTNAMELEN
-	return MAXHOSTNAMELEN;
-#elif HOST_NAME_MAX
-	return HOST_NAME_MAX;
-#endif
-	return 64;
-}
-
-/*
  * The usleep function was marked obsolete in POSIX.1-2001 and was removed
  * in POSIX.1-2008.  It was replaced with nanosleep() that provides more
  * advantages (like no interaction with signals and other timer functions).
